contracts/staking/contracts/src/Staking.sol

```diff
@@ -66,4 +74,298 @@ contract Staking is
             _zrxVaultAddress
         );
     }
+
+    /// @dev This function will fail if the storage layout of this contract deviates from
+    ///      the original staking contract's storage. The use of this function provides assurance
+    ///      that regressions from the original storage layout will not occur.
+    function _assertStorageLayout()
+        internal
+        pure
+    {
+        assembly {
+            let slot := 0x0
+            let offset := 0x0
+
+            /// Ownable
+
+            assertSlotAndOffset(
```

Comment:
Totally optional, but it could be more readable to make an array where each element has a state variable's slot/offset. Then loop through, do the check in assembly and revert on failure in regular solidity.